### PR TITLE
Add missing generator parts

### PR DESCRIPTION
**Pull request by ramizdundar**
Some files were missing from generators, I added them and extended the tests. ~`asserMapStoresAreEqual()` includes what compatibility check doesn't include.~ Also extended `MapConfigChecker.isCompatible(MapStoreConfig, MapStoreConfig)`.

Fixes https://github.com/hazelcast/hazelcast-enterprise/issues/4777

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/internal/dynamicconfig/AbstractDynamicConfigGeneratorTest.java [hunk 1]
```diff
@@ -915,6 +915,15 @@ private void testMap(MapStoreConfig mapStoreConfig) {
         assertEquals(attrConfig.getName(), decAttrConfig.getName());
         assertEquals(attrConfig.getExtractorClassName(), decAttrConfig.getExtractorClassName());
         ConfigCompatibilityChecker.checkMapConfig(expectedConfig, actualConfig);
+
+        assertMapStoresAreEqual(expectedConfig.getMapStoreConfig(), actualConfig.getMapStoreConfig());
```
Comment (kwart) [inline hazelcast/src/test/java/com/hazelcast/internal/dynamicconfig/AbstractDynamicConfigGeneratorTest.java:919]: Shouldn't this rather be part of the `com.hazelcast.config.ConfigCompatibilityChecker.MapConfigChecker.isCompatible(MapStoreConfig, MapStoreConfig)`?

**Review by kwart — COMMENTED** [review]
One comment/question - otherwise looks good.

File: hazelcast/src/test/java/com/hazelcast/internal/dynamicconfig/AbstractDynamicConfigGeneratorTest.java (hunk 1 above)
Comment (ramizdundar) [reply hazelcast/src/test/java/com/hazelcast/internal/dynamicconfig/AbstractDynamicConfigGeneratorTest.java:919]: I am not sure, I thought the reason we use that method is map configs can be different across the cluster as long as compatible (This is not possible for dynamic configs but it's for static configs I believe). And maybe these can be different between map stores hence, not included.

I also tried only using equals, but in that case class equalities create a problem in tests. So we need the logic in `com.hazelcast.config.ConfigCompatibilityChecker.MapConfigChecker.isCompatible` for checking equality of classes but I don't know/think that method needs/can have excluded assertions.

Comment (ramizdundar) [reply hazelcast/src/test/java/com/hazelcast/internal/dynamicconfig/AbstractDynamicConfigGeneratorTest.java]: https://github.com/hazelcast/hazelcast/pull/20743/commits/737da411c3009049b39e8cc0b598d7869dc0130d

**Review by kwart — APPROVED** [review]

**Comment by olukas** [comment]
run-lab-run

**Comment by ramizdundar** [comment]
run-lab-run

**Comment by olukas** [comment]
Force merged after discussion with @AyberkSorgun

**Pull request merged**